docker/docker-entrypoint.sh

```diff
@@ -14,6 +14,13 @@ then
     set -- /bin/consul-terraform-sync "$@"
 fi
 
+# If user is trying to run consul-terraform-sync with no arguments (daemon-mode),
+# then $1 will default to '/bin/sh'. In this case, do not pass along default arguments.
+if [ "$1" = '/bin/sh' ]
```

Comment:
This should check for more than `$1` equating `/bin/sh` as it is a very common pattern to run the container with `/bin/sh` in order to debug issues. For example..

```
docker run -ti --rm --network host hashicorp/consul-terraform-sync:0.1.0-beta /bin/sh
```

So you don't want to test for `/bin/sh`.. at least by itself. The full command when the docker container is run without arguments is `/bin/sh -c /bin/${NAME}`.  So I suggest a matching the full string here..

```
if test "$*" = '/bin/sh -c /bin/${NAME}'
then ...
```

Reply:
Note I tweaked the above several times... so don't go off the email version.

Reply:
Thanks for catching that and sharing that common use-case! That's super helpful to know. I committed your suggested change and updated the comment as well.